mlflow/server/handlers.py

```diff
@@ -3134,16 +3162,32 @@ def get_trace_artifact_handler():
             error_code=BAD_REQUEST,
         )
 
+    trace_data_in_artifact_repo = False
     try:
-        traces = _get_tracking_store().batch_get_traces([request_id], None)
-        if len(traces) != 1:
-            raise MlflowException(
-                "Failed to get trace data from tracking store. Please check the request_id."
-            )
-        trace_data = traces[0].data.to_dict()
-    # For stores that don't support batch get traces, or if the trace data is not stored in the
-    # tracking store, we need to get the trace data from the artifact repository.
-    except (MlflowTracingException, MlflowNotImplementedException):
+        # allow partial so the front end can render in-progress traces
+        trace = _get_tracking_store().get_trace(request_id, allow_partial=True)
+        trace_data = trace.data.to_dict()
+    except MlflowNotImplementedException:
+        pass
+    except MlflowTracingException:
+        trace_data_in_artifact_repo = True
+
+    if not trace_data_in_artifact_repo:
+        try:
+            traces = _get_tracking_store().batch_get_traces([request_id], None)
+            if len(traces) != 1:
+                raise MlflowException(
+                    f"Trace with id={request_id} not found, please check if the "
+                    "trace ID is correct.",
```

Comment:
```suggestion
                    f"Trace with id={request_id} not found",
```

I think `please check ...` is redundant.